### ISSUE TITLE
No physical device 

### ISSUE THREAD

**Issue by isi8787** [opentrace-community/opentrace-android#29]
Hi Everyone,

I currently dont have a physical device and was setting up just using android studio. Given the importance of bluetooth this was an immediate show stopper during setup (I made it past the otp screen after setting up in firebase).
I have tried launching the app using an android image on virtual box but again i am having issues with bluetooth. In both cases after the OTP screen where i am asked to give permission to access bluetooth the app resets to first screen.
Has anybody been able to work with the app without a physical device? If so what was your setup?

Thanks

**Comment by alexissinglaire** [comment]
@isi8787 : i dont think you can run this app in the emulator device. coz the firebase will give warning and unable to operate in emulated device. i believe there is no other solution for your concern.

**Comment by erwanmace** [comment]
@isi8787 the Android simulator does not support Bluetooth (BluetoothAdapter.getDefaultAdapter() would return null).
The following post on SO will however show you how to get BLE support by using Android in a VMware or VirtualBox virtual machine:
https://stackoverflow.com/questions/20348743/bluetooth-low-energy-on-android-emulator/27712017#27712017
